test/distributed/test_c10d_nccl.py

```diff
@@ -852,6 +852,30 @@ def test_comm_split_subgroup(self):
         self.assertEqual(tensor, original_tensor)
         dist.destroy_process_group()
 
+    @requires_nccl_version((2, 18), "Need NCCL 2.18+ for ncclCommSplit")
+    @skip_but_pass_in_sandcastle_if(not TEST_MULTIGPU, "NCCL test requires 2+ GPUs")
+    def test_comm_eager_init_subgroup(self):
+        # Test `ncclCommSplit` for smaller subgroups of the world when
+        # we've passed a specific device_id to init_process_group.
+        store = c10d.FileStore(self.file_name, self.world_size)
+        device = torch.device(f"cuda:{self.rank}")
+        # default PG comm is not initialized yet
+        pg = self._create_process_group_nccl(store, self.opts())
+        backend = pg._get_backend(torch.device(device))
+        self.assertEqual(backend._is_initialized(), False)
+
+        tensor = torch.full((1,), self.rank).cuda(device)
+        new_group = c10d.new_group([0, 1], device_id=device)
+        self.assertEqual(backend.comm_split_count(), 0)
```

Comment:
nit: I propose we stop using `comm_split_count` for testing.

Reply:
This one is widely used on split related tests, any alternatives?

Reply:
No alternatives. Just proposing that we stop using it.

Reply:
Then, not using it is basically equivalent to removing valid python tests and safety guards to future code changes. Unless we have an alternative, e.g., fully trusted c++ tests (but still missing the e2e py test)